src/main/java/uk/gov/companieshouse/service/dissolution/chips/ChipsResponseService.java

```diff
@@ -38,6 +51,14 @@ public void saveAndNotifyDissolutionApplicationOutcome(ChipsResponseCreateReques
         dissolution.setVerdict(dissolutionVerdict);
         dissolution.setActive(false);
 
+        if (dissolutionVerdict.getResult() == VerdictResult.REJECTED) {
+            Optional<RefundResponse> refundResponse = paymentsClient.refundPayment(dissolution.getPaymentInformation().getReference());
+            if (refundResponse.isPresent()) {
+                RefundInformation refund = refundInformationMapper.mapToRefundInformation(refundResponse.get());
+                dissolution.getPaymentInformation().setRefund(refund);
+            }
+        }
```

Comment:
feels like this could be abstracted behind a `DissolutionRefundService` i.e.
```
if (dissolutionVerdict.getResult() == VerdictResult.REJECTED) {
  dissolutionRefundService.refundDissolution(dissolution);
}
```

Then the DissolutionRefundService can:
1. Map to the refund request body
2. Initiate the refund via the Payments client
3. Update the dissolution

Reply:
We should also feature toggle this i.e.
```
if (refundsEnabled && dissolutionVerdict.getResult() == VerdictResult.REJECTED) {
  dissolutionRefundService.refundDissolution(dissolution);
}
```

Reply:
I figured that doing this in with the existing dissolutions modifying would reduce db calls.
Is this not a reasonable consideration?